athena-msk/athena-msk.yaml

```diff
@@ -0,0 +1,125 @@
+Transform: 'AWS::Serverless-2016-10-31'
```

Comment:
How did you guys test this with IAM authentication?

It looks like you will have to add something to this template in order for the user to specify a custom function execution role with the correct permissions otherwise it won't work, right?

See this: https://docs.aws.amazon.com/lambda/latest/dg/with-msk.html#msk-permissions-iam-policy

I think (not sure), that you'll have to do something like what was done in the athena-synapse connector with setting up a custom FunctionRole: https://github.com/awslabs/aws-athena-query-federation/blob/020df2a851250054b455d900bc6d0122b02e1b66/athena-synapse/athena-synapse.yaml#L87

Except in this case, you will have to add `Parameters` for the the various `arn:aws:kafka:___` resources that the user wants to specify to be a part of the policy document.

Reply:
Note that I could be mistaken about this btw and IAM just works with the template as is already.

If so, can you guys just provide a brief explanation as to why the function execution policy isn't necessary?

Thanks.

Reply:
Hi @henrymai , For IAM, it is a custom role that need to be attached to the lambda execution role. User has to create a policy [Go to MSK cluster -> Properties ->Security Settings -> Configure IAM policy -> fetch the content and create a new policy]and attach that to lambda execution role.

Reply:
Ok, in that situation, can we do something like what was done with Synapse so that the user can specify the Function Execution Role up front as a parameter?

Reply:
Sure Henry, We can do that, Basically if auth type is IAM, user must provide the custom lambda role. This will be part of readme and yaml.  If he does not provide custom lambda role, later on he has to add the custom policy to lambda execution role manually. Is that a fair understanding?

Reply:
Yes, that's fine for now I think.

Reply:
Sure, will make this change available.